arraycontext/impl/pytato/compile.py

```diff
@@ -154,7 +154,34 @@ def id_collector(keys, ary):
     return pmap(arg_id_to_arg), pmap(arg_id_to_descr)
 
 
-def _get_f_placeholder_args(arg, kw, arg_id_to_name):
+def _transform_array(ary, actx):
+    """
+    Returns the transformed version of *ary* after calling
+    :meth:`ArrayContext.transform_dag` on it.
+    """
+    import pyopencl.array as cla
+    from arraycontext.impl.pyopencl.taggable_cl_array import (to_tagged_cl_array,
+                                                              TaggableCLArray)
+    if isinstance(ary, pt.Array):
+        dag = pt.make_dict_of_named_arrays({"_actx_out": ary})
+        return actx.transform_dag(dag)["_actx_out"].expr
```

Comment:
Add a comment that this is about metadata propagation.

Reply:
Done!